lms/djangoapps/edxnotes/decorators.py

```diff
@@ -26,7 +26,7 @@ def get_html(self, *args, **kwargs):
         )
 
         runtime = getattr(self, 'descriptor', self).runtime
-        if not hasattr(runtime, 'modulestore'):
+        if not hasattr(runtime, 'modulestore') or getattr(runtime, "disable_edxnotes_for_test", False):
```

Comment:
Question: Why was this necessary to add?

Reply:
Good question. I didn't really like adding this so perhaps there is a better way. In fact I didn't actually figure out why the test in question was previously passing. Hopefully I am missing something obvious.

The problem I was facing was [this test failure](https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/32880/testReport/junit/xmodule.xmodule.tests.test_conditional/ConditionalBlockXmlTest/Run_Tests___commonlib_unit___test_conditional_module/) (in `test_conditional.ConditionalBlockXmlTest.test_conditional_module`):

```
>       ajax = json.loads(module.handle_ajax('', ''))

common/lib/xmodule/xmodule/tests/test_conditional.py:304: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
common/lib/xmodule/xmodule/conditional_module.py:291: in handle_ajax
    fragments = [child.render(STUDENT_VIEW).to_dict() for child in self.get_display_items()]
common/lib/xmodule/xmodule/conditional_module.py:291: in <listcomp>
    fragments = [child.render(STUDENT_VIEW).to_dict() for child in self.get_display_items()]
../edx-venv-3.8/edx-venv/lib/python3.8/site-packages/xblock/core.py:198: in render
    return self.runtime.render(self, view, context)
common/lib/xmodule/xmodule/x_module.py:1970: in render
    return self.__getattr__('render')(block, view_name, context)
common/lib/xmodule/xmodule/x_module.py:1419: in render
    return super().render(block, view_name, context=context)
../edx-venv-3.8/edx-venv/lib/python3.8/site-packages/xblock/runtime.py:826: in render
    frag = view_fn(context)
common/lib/xmodule/xmodule/html_module.py:92: in student_view
    fragment = Fragment(self.get_html())
lms/djangoapps/edxnotes/decorators.py:40: in get_html
    user = self.runtime.get_real_user(self.runtime.anonymous_student_id)
```

Basically the test of a `conditional` module is using a mostly-mocked runtime, and when it tries to (mock) render an HTML XBlock child in that runtime, the HTML XBlock is being decorated by `edxnotes` which overrides the `html` block's `get_html()` method and tries to [call methods on the runtime which don't exist](https://github.com/edx/edx-platform/blob/b7a34887c234d6598fe53186f90293e66450e886/lms/djangoapps/edxnotes/decorators.py#L40) because this test uses a mock runtime.

It seems like there should be some simple mechanism for disabling `edxnotes` entirely since it's irrelevant to the test in question, but [the check that disables edxnotes](https://github.com/edx/edx-platform/blob/b7a34887c234d6598fe53186f90293e66450e886/lms/djangoapps/edxnotes/decorators.py#L42) actually comes after the line that's causing the error. Hence I added another mechanism to disable it.

Alternately I could just add mocks for `anonymous_student_id` and `get_real_user` into the mock runtime - that may be cleaner? Or is there a better way to disable edxnotes?

Reply:
Instead of the `getattr(runtime, "disable_edxnotes_for_test", False)`, maybe you could put `settings.FEATURES.get("ENABLE_EDXNOTES")`? It's still a little hacky in that there are other things that should be able to disable this feature, but at least it would give a clear way to to disable it via conventional settings override.

Reply:
Ah yes, that makes way more sense. Done in 278c839e683eb09e317d053456a7cfccfea80492